core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1176,6 +1187,20 @@ public Builder removeBranch(String branch) {
       return this;
     }
 
+    public Builder setStatistics(StatisticsFile statisticsFile) {
```

Comment:
The method name `setStatistics` doesn't quite describe what is happening. It seems to imply that the statistics are replaced by the file passed in, but this is actually only replacing the statistics for a particular snapshot. I think `setStatistics(long snapshotId, StatisticsFile statisticsFile)` is better because it is clear you're setting the stats for an ID. That also matches `removeStatistics(long snapshotId)`.